### PR TITLE
Getting rid of bare excepts where possible

### PR DESCRIPTION
**Pull request by ayodejiige**
The last exception in _print_RandomDomain() should be called when the
RandomDomain object doesn't have a set property which will raise an
IndexError

<If this pull request fixes an issue please indicate which issue by typing "Fixes #NNNN" below.>
This is a first commit for #11668

### CODE REVIEW

**Comment by asmeurer** [comment]
Wouldn't the second case also raise AttributeError?

**Comment by ayodejiige** [comment]
It raises an index error because the RandomDomain class and child classes have a set property.
` @property
    def set(self):
        return self.args[1]`
The index error will be raised in cases where self.args[1] doesn't exist.

**Comment by ayodejiige** [comment]
I did notice that I wouldn't this case though:
```python
class ConditionalDomain(RandomDomain):
    ...
    @property
    def set(self):
        raise NotImplementedError("Set of Conditional Domain not Implemented")
   ...
```
I will add the code to catch that. Also, this is my first open source project and I was wondering if issues on this projects are typically fixed with one pull request and if it's fine to continuously push to a pull request.

**Comment by asmeurer** [comment]
Please push to the same pull request.

**Comment by asmeurer** [comment]
It would be better to refactor the code to not use try-except logic at all. The fact that `set` raises IndexError is not at all obvious and could change if the implementation changes. It would be better to do `hasattr` checks.

**Comment by ayodejiige** [comment]
I will make the necessary changes. Thanks!

**Comment by gxyd** [comment]
@asmeurer can you please see if the changes are apt? Seems like he did 'try' address the requested changes.

**Comment by gxyd** [comment]
@nilabja10201992 would you like to review this pull request?

**Comment by nilabja10201992** [comment]
@gxyd sure Sure I'll review them btw should I work on the fixes that you mentioned yesterday?

**Comment by gxyd** [comment]
Yes, you can work on that as well.

**Comment by nilabja10201992** [comment]
Okay :)

File: sympy/printing/pretty/pretty.py
Comment (asmeurer) [inline sympy/printing/pretty/pretty.py]: I believe this should be `self._print(None)` to match what was there before.

File: sympy/printing/str.py
Comment (asmeurer) [inline sympy/printing/str.py]: Printers should not return None. The above case should be the `else` case.

File: sympy/printing/latex.py
Comment (asmeurer) [inline sympy/printing/latex.py]: Same here.

**Review by asmeurer — CHANGES_REQUESTED** [review]

**Comment by nilabja10201992** [comment]
@gxyd seems correct to me, as @asmeurer stated _else shouldn't return None_, otherwise, everything seems okay.

**Comment by gxyd** [comment]
@ayodejiige can you please address the requested changes?

**Comment by gxyd** [comment]
@nilabja10201992 would you like to make changes here as suggested by Aaron?
You would need to add 'remote' here. The main purpose could be figuring out how to push to some other 'remote', search for it over the web.

In any case, let me know if you need help.

**Comment by nilabja10201992** [comment]
@gxyd I've made changes to my local branch but cannot push to the remote branch. It says access denied.

**Comment by gxyd** [comment]
I'll get back in the morning.

**Comment by gxyd** [comment]
Can you paste output of `git remote -v` (running in SymPy root folder) here?

**Comment by nilabja10201992** [comment]
```
origin	https://github.com/nilabja10201992/sympy (fetch)
origin	https://github.com/nilabja10201992/sympy (push)
pb	https://github.com/ayodejiige/sympy (fetch)
pb	https://github.com/ayodejiige/sympy (push)
upstream	https://github.com/sympy/sympy (fetch)
upstream	https://github.com/sympy/sympy (push)
```

**Comment by gxyd** [comment]
And first you do `git remote fetch ayodejiige`. And after that you do `git checkout -b 11668_bare_excepts ayodejiige/11668_bare_excepts` (may be you **didn't** execute this command)?

And make changes to `11668_bare_excepts` **not** to `ayodejiige/11668_bare_excepts`. And finally you do `git push ayodejiige 11668_bare_excepts` (may be you did this right).

**Comment by ylemkimon** [comment]
@gxyd Only those with write access to main branch(`sympy/sympy`) can push commits to PR branch.

**Comment by nilabja10201992** [comment]
@gxyd when I'm trying to push using `git it push ayodejiige 11668_bare_excepts` I'm getting these errors
```
error: src refspec 11668_bare_excepts does not match any.
error: failed to push some refs to 'ayodejiige'

```

**Comment by gxyd** [comment]
Okay, yes you can't do that. I think that can be done via `git cherry-pick ...` command https://git-scm.com/docs/git-cherry-pick . That would be sending in a new pull request, not pushing to this branch.

**Comment by nilabja10201992** [comment]
@gxyd can you please look at the changes here https://github.com/sympy/sympy/compare/master...nilabja10201992:pb?expand=1 , should I pull request?

**Comment by gxyd** [comment]
I have fixed the requested changes. Will merge it when it passes the test.

**Comment by gxyd** [comment]
Tests pass. Merging.

**Pull request merged**
